Refactor local lint

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58798 Refactor local lint**

In #58623 there was a bug in `make quicklint` where ShellCheck would run on the entire repo when there were no files. This PR fixes that by refactoring out common stuff (like skipping quicklint when there are no files, let checks do their own file filtering) and pushes the logic into a runner class.

Differential Revision: [D28649889](https://our.internmc.facebook.com/intern/diff/D28649889)

driazati committed:
Refactor local lint

In #58623 there was a bug in `make quicklint` where ShellCheck would run on the entire repo when there were no files. This PR fixes that by refactoring out common stuff (like skipping quicklint when there are no files) and pushes the logic into a runner class.

[ghstack-poisoned]

samestep commented:
btw what's the test plan for this? I'm assuming
```
python tools/test/test_actions_local_runner.py
```

samestep reviewed (approved):
looks great, thank you! just a few nits

driazati committed:
Update on "Refactor local lint"


In #58623 there was a bug in `make quicklint` where ShellCheck would run on the entire repo when there were no files. This PR fixes that by refactoring out common stuff (like skipping quicklint when there are no files, let checks do their own file filtering) and pushes the logic into a runner class.

[ghstack-poisoned]

samestep commented on tools/actions_local_runner.py:
```diff
@@ -117,176 +125,191 @@ async def shell_cmd(
-        # flake8 will still use .flake8 to filter this list by the 'exclude's
-        # in the config
-        cmd += files
+        files = self.filter_ext(files, {"py"})
```
you need the `.`:
```suggestion
        files = self.filter_ext(files, {".py"})
```

driazati replied:
Nice catch, I thought the GitHub UI commit buttons would break ghstack but it looks like everything turned out fine

samestep replied:
oh weird! I also thought they would break `ghstack`, since I'm pretty sure I've seen that happen in the past... strange

driazati replied:
Just kidding it did break it (and silently too!), it shows the changes applied on the GitHub UI but `ghstack checkout` doesn't have them nor does the `ghimport`ed diff

samestep replied:
oh sad :cry:

samestep commented on tools/actions_local_runner.py:
```diff
@@ -117,176 +125,191 @@ async def shell_cmd(
+    name = "mypy (skipped typestub generation)"
+
+    def filter_files(self, files: List[str]) -> List[str]:
+        return self.filter_ext(files, {"py", "pyi"})
```
same here:
```suggestion
        return self.filter_ext(files, {".py", ".pyi"})
```

samestep commented on tools/actions_local_runner.py:
```diff
@@ -117,176 +125,191 @@ async def shell_cmd(
-            ["tools/run_shellcheck.sh"] + [
-                os.path.join(REPO_ROOT, f) for f in files
+    def filter_files(self, files: List[str]) -> List[str]:
+        return self.filter_ext(files, {"sh"})
```
and here:
```suggestion
        return self.filter_ext(files, {".sh"})
```

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28649889).

driazati committed:
Apply suggestions from code review

Co-authored-by: Sam Estep <sestep@fb.com>

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28649889).

driazati committed:
Update on "Refactor local lint"


In #58623 there was a bug in `make quicklint` where ShellCheck would run on the entire repo when there were no files. This PR fixes that by refactoring out common stuff (like skipping quicklint when there are no files, let checks do their own file filtering) and pushes the logic into a runner class.

Differential Revision: [D28649889](https://our.internmc.facebook.com/intern/diff/D28649889)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28649889).
